recipes/default.rb

```diff
@@ -209,6 +209,17 @@
   EOH
 end
 
+bash 'create_libhopsnvml_symlink' do
+  user "root"
+  code <<-EOH
+    set -e
+    if [ ! -f #{node.hops.base_dir}/lib/native/libhopsnvml.so ]; then
+      ln -s #{node.hops.base_dir}/share/hadoop/yarn/lib/libhopsnvml-1.0.so #{node.hops.base_dir}/lib/native/libhopsnvml.so
```

Comment:
You should have a version for libhopsnvml.
It should be like:
libhopsnvml-#{node.hops.nvml_version}.so
And then add 
default.hops.nvml_version = "1.0"
to attributes/default.rb

Also, it's easier to run this code with:
user node.hops.hdfs.user
group node.hops.group
(instead of user "root")
Then you can remove the chown statement.